### PR TITLE
Simple CIFAR experiments

### PR DESCRIPTION
**Pull request by subutai**

### CODE REVIEW

File: nupic/research/support/parse_config.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        try:
+          item[k] = eval(v)
+        except (NameError, SyntaxError):
+          item[k] = v
+
+      params[exp] = item
+
+  return params
+
+
```
Comment (lscheinkman) [inline nupic/research/support/parse_config.py]: Great idea to reuse this code!

File: projects/whydense/cifar/cifar_tune.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                         )),
+    trial_executor=config.get("trial_executor", None),
+    checkpoint_at_end=config.get("checkpoint_at_end", False),
+    checkpoint_freq=config.get("checkpoint_freq", 0),
+    resume=config.get("resume", False),
+    reuse_actors=config.get("reuse_actors", False),
+    verbose=config.get("verbose", 0)
+  )
+
+def parse_config(config_file, experiments=None):
```
Comment (lscheinkman) [inline projects/whydense/cifar/cifar_tune.py]: Should use 'nupic.research.support.parse_config' instead

File: projects/whydense/cifar/list_experiments.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    """Summarizes trials in the directory subtree starting at the given path."""
+    browser = experimentBrowser(experiment_path)
+    p = browser.get_value(exp_substring=name, tag=tag, which=which)
+    p.sort_values(tag, axis=0, ascending=False,
+                 inplace=True, na_position='last')
+    print(p)
+
+
+if __name__ == "__main__":
+    summarize_trials()
```
Comment (lscheinkman) [inline projects/whydense/cifar/list_experiments.py]: ray 0.6.6 just added a few utilities to visualize the results. 
https://github.com/ray-project/ray/blob/master/python/ray/tune/visual_utils.py

File: nupic/research/frameworks/pytorch/cifar_experiment.py [hunk 4]
```diff
@@ -20,54 +20,70 @@
...
 
 
-class TinyCIFAR(tune.Trainable):
+class TinyCIFAR(object):
   """
-  ray.tune trainable class:
-  - Override _setup to reset the experiment for each trial.
-  - Override _train to train and evaluate each epoch
-  - Override _save and _restore to serialize the model
+  Generic class for creating tiny CIFAR models. This can be used with Ray tune
```
Comment (lscheinkman) [inline nupic/research/frameworks/pytorch/cifar_experiment.py]: Nice! Maybe later we could refactor into a generic interface for any tunable experiment, not just CIFAR

Comment (subutai) [reply nupic/research/frameworks/pytorch/cifar_experiment.py]: Good idea. We should do that (in a separate PR).

File: nupic/research/frameworks/pytorch/cifar_experiment.py [hunk 5]
```diff
@@ -112,20 +132,86 @@ def _setup(self, config):
                                      transform=self.transform_train)
 
     self.train_loader = torch.utils.data.DataLoader(
-      train_dataset, batch_size=batch_size, shuffle=True)
+      train_dataset, batch_size=batch_size, shuffle=True,
```
[the remark this answers is not in the document]
Comment (subutai) [reply nupic/research/frameworks/pytorch/cifar_experiment.py]: Fixed.

File: nupic/research/frameworks/pytorch/cifar_experiment.py [hunk 6]
```diff
@@ -112,20 +132,86 @@ def _setup(self, config):
...
+      train_loader = self.train_loader
+      batches_in_epoch = self.batches_in_epoch
+
+    self._preEpoch()
+    trainModel(model=self.model, loader=train_loader,
+               optimizer=self.optimizer, device=self.device,
+               batches_in_epoch=batches_in_epoch,
+               criterion=self.loss_function)
+    self.model.apply(rezeroWeights)
+    self.model.apply(updateBoostStrength)
```
[the remark this answers is not in the document]
Comment (subutai) [reply nupic/research/frameworks/pytorch/cifar_experiment.py]: Good idea - fixed.

File: nupic/research/frameworks/pytorch/cifar_experiment.py [hunk 7]
```diff
@@ -112,20 +132,86 @@ def _setup(self, config):
...
+    trainTime = time.time() - t1
+
+    ret = self._runNoiseTests(self.noise_values, self.test_loaders)
```
[the remark this answers is not in the document]
Comment (subutai) [reply nupic/research/frameworks/pytorch/cifar_experiment.py]: Yes, for now I do like to do this. We can add additional options for not doing it every epoch later.

File: projects/whydense/cifar/cifar_tune.py (hunk 2 above)
Comment (subutai) [reply projects/whydense/cifar/cifar_tune.py]: I would like to, but the 'eval' didn't work for tune.grid_search. I think maybe it needs to be in the same scope as the parse_config function...??

File: projects/whydense/cifar/list_experiments.py (hunk 3 above)
Comment (subutai) [reply projects/whydense/cifar/list_experiments.py]: Cool!

**Pull request merged**